cmd/pbackend/statuser.go

```diff
@@ -153,16 +152,22 @@ func checkSourceAvailabilityAWS(ctx context.Context) {
 				Identity:     s.Identity,
 				ResourceType: "Application",
 			}
-			_, err = clients.GetEC2Client(ctx, &s.Authentication, "")
+			ec2Client, err := clients.GetEC2Client(ctx, &s.Authentication, "")
 			if err != nil {
 				sr.Status = kafka.StatusUnavailable
 				sr.Err = err
 				logger.Warn().Err(err).Msg("Could not get aws assumed client")
```

Comment:
Shouldn't we make sure, that when `GetEC2Client` errors out, we do not call CheckPermission on it? that might be nil pointer reference, right?

I'd just return here honestly, we would need to bump the metric here as well, but that's just three extra lines (write to channel, bump metric and return), not a big deal. WDYT?

Reply:
I might be missing something here tho :)

Reply:
That's definitely true, we should return here! TY.

Reply:
Guys, please no `return` statements here, this is a channel processing loop!

One note: this is one of rare examples when `goto` statement which is [available in Go](https://go.dev/ref/spec#Goto_statements) could be reasonable, also a function could be good fit. I see some code being duplicated (creating the result, sending the result to).

Reply:
I believe @avitova solved it pretty well :)
Good compromise! :+1:

Reply:
Oh no, did I scared you, @ezr-ondrej ? :-)

You merge me a trigger and stored SQL procedure and the next day I debate loops :-D